hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SqlPojoTest.java

```diff
@@ -423,6 +423,20 @@ public void test_nestedField() {
                 .hasMessageContaining("Invalid external name: this.owner.name");
     }
 
+    @Test
+    public void when_noFieldsResolved_then_wholeValueMapped() {
+        String name = randomName();
+
+        sqlService.execute(javaSerializableMapDdl(name, Object.class, Object.class));
+
+        Person key = new Person(1, "foo");
+        Person value = new Person(2, "bar");
+        instance().getMap(name).put(key, value);
+
+        assertRowsAnyOrder("SELECT __key, this FROM " + name,
+                singletonList(new Row(key, value)));
+    }
+
```

Comment:
The following test if added to this class fails:
```java
    @Test
    public void when_keyHasKeyField_then_wholeKeyResolved() {
        String name = randomName();
        sqlService.execute(javaSerializableMapDdl(name, ClassWithKey.class, Integer.class));
        instance().getMap(name).put(new ClassWithKey(), 0);
        assertRowsAnyOrder("SELECT * FROM " + name,
                singletonList(new Row(new ClassWithKey(), 0)));
    }

    public static class ClassWithKey implements Serializable {
        public int __key;

        @Override
        public boolean equals(Object o) {
            if (this == o) {
                return true;
            }
            if (o == null || getClass() != o.getClass()) {
                return false;
            }
            ClassWithKey that = (ClassWithKey) o;
            return __key == that.__key;
        }

        @Override
        public int hashCode() {
            return Objects.hash(__key);
        }
    }
```

The `ClassWithKey` has a `__key` field. What happened is that we mapped the key's field under the `__key` name, and the whole key isn't mapped.

The point of this PR is to make sure that `__key` *always* maps the whole key. It's required for the DELETE functionality.